src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -5,146 +5,1406 @@
+import { Action2, MenuId } from 'vs/platform/actions/common/actions';
+import { KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
+import { resolveResourcesForSearchIncludes } from 'vs/workbench/services/search/common/queryBuilder';
+import { getMultiSelectedResources, IExplorerService } from 'vs/workbench/contrib/files/browser/files';
+import { IFileService } from 'vs/platform/files/common/files';
+import { IWorkspaceContextService } from 'vs/platform/workspace/common/workspace';
+import { ExplorerFolderContext, ExplorerRootContext, FilesExplorerFocusCondition, VIEWLET_ID as VIEWLET_ID_FILES } from 'vs/workbench/contrib/files/common/files';
+import { getWorkspaceSymbols, IWorkspaceSymbol, SearchStateKey, SearchUIState } from 'vs/workbench/contrib/search/common/search';
+import { IPaneCompositePartService } from 'vs/workbench/services/panecomposite/browser/panecomposite';
+import { ExplorerViewPaneContainer } from 'vs/workbench/contrib/files/browser/explorerViewlet';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { IQuickInputService } from 'vs/platform/quickinput/common/quickInput';
+import { ToggleCaseSensitiveKeybinding, TogglePreserveCaseKeybinding, ToggleRegexKeybinding, ToggleWholeWordKeybinding } from 'vs/editor/contrib/find/browser/findModel';
+
+/** INTERFACES */
+interface ISearchActionContext {
+	readonly viewer: WorkbenchCompressibleObjectTree<RenderableMatch>;
+}
 
-export function isSearchViewFocused(viewsService: IViewsService): boolean {
-	const searchView = getSearchView(viewsService);
-	const activeElement = document.activeElement;
-	return !!(searchView && activeElement && DOM.isAncestor(activeElement, searchView.getContainer()));
+export interface IMatchActionContext extends ISearchActionContext {
+	readonly element: Match;
 }
 
-export function appendKeyBindingLabel(label: string, inputKeyBinding: number | ResolvedKeybinding | undefined, keyBindingService2: IKeybindingService): string {
-	if (typeof inputKeyBinding === 'number') {
-		const keybinding = createKeybinding(inputKeyBinding, OS);
-		if (keybinding) {
-			const resolvedKeybindings = keyBindingService2.resolveKeybinding(keybinding);
-			return doAppendKeyBindingLabel(label, resolvedKeybindings.length > 0 ? resolvedKeybindings[0] : undefined);
+export interface IFileMatchActionContext extends ISearchActionContext {
+	readonly element: FileMatch;
+}
+
+export interface IFolderMatchActionContext extends ISearchActionContext {
+	readonly element: FolderMatch;
+}
+
+export interface IFindInFilesArgs {
+	query?: string;
+	replace?: string;
+	preserveCase?: boolean;
+	triggerSearch?: boolean;
+	filesToInclude?: string;
+	filesToExclude?: string;
+	isRegex?: boolean;
+	isCaseSensitive?: boolean;
+	matchWholeWord?: boolean;
+	useExcludeSettingsAndIgnoreFiles?: boolean;
+	onlyOpenEditors?: boolean;
+}
+
+/** ACTIONS */
+
+/** ACTIONS: REPLACE/REMOVE RESULTS */
```

Comment:
Use `/*` for block comments and `/**` for jsdoc. As-is, this comment shows up in the hover for RemoveAction

Reply:
You might be interested in the `//#region` syntax as well